### PR TITLE
[#180] Created Alerts database schema and mock data

### PR DESCRIPTION
**Pull request by Tiernebre**
## Changes
- New alert database schema has been created.

### CODE REVIEW

File: docker/postgres/docker-entrypoint-initdb.d/mock-data/interfaith_DDL.sql [hunk 1]
```diff
@@ -54,22 +50,25 @@ CREATE TABLE needs (
...
+  calloutneed_id BIGINT REFERENCES users NOT NULL,
   -- donor_id BIGINT REFERENCES donors NOT NULL
 );
 
--- transaction stuff, activations, 
+CREATE TABLE alerts (
+  alert_id BIGSERIAL PRIMARY KEY,
+  alert_sent_date TIMESTAMP NULL,
+  alert_created_date TIMESTAMP DEFAULT CURRENT_TIMESTAMP
+);
```
Comment (TylerNicholls) [inline docker/postgres/docker-entrypoint-initdb.d/mock-data/interfaith_DDL.sql:63]: Is created_date a necessary field? It obviously doesn't hurt us to capture it so no worries either way just curious.

Another possible field: alert_text/description? So she can say "Click on the app now to check out what we need for this winter" or something along those lines? Could always add it now and remove the field later, which seems easier than trying to add it later?

File: docker/postgres/docker-entrypoint-initdb.d/mock-data/interfaith_DDL.sql [hunk 2]
```diff
@@ -54,22 +50,25 @@ CREATE TABLE needs (
...
   donation_date TIMESTAMP DEFAULT CURRENT_TIMESTAMP,
-  needcallout_id BIGINT REFERENCES users NOT NULL,
+  calloutneed_id BIGINT REFERENCES users NOT NULL,
   -- donor_id BIGINT REFERENCES donors NOT NULL
 );
 
--- transaction stuff, activations, 
+CREATE TABLE alerts (
+  alert_id BIGSERIAL PRIMARY KEY,
+  alert_sent_date TIMESTAMP NULL,
```
Comment (TylerNicholls) [inline docker/postgres/docker-entrypoint-initdb.d/mock-data/interfaith_DDL.sql:61]: I'm assuming sent_date is the published date? Should it be nullable?

**Review by TylerNicholls — COMMENTED** [review]
If you think it's fine like this I wouldn't be opposed to approving it in this state, just some general implementation type questions

File: docker/postgres/docker-entrypoint-initdb.d/mock-data/interfaith_DDL.sql (hunk 1 above)
Comment (Tiernebre) [reply docker/postgres/docker-entrypoint-initdb.d/mock-data/interfaith_DDL.sql:63]: Created date is not necessary, it's just really easy to implement and gives us more information about an alert.

Good call on the alert_text / alert_description! I can get that added.

File: docker/postgres/docker-entrypoint-initdb.d/mock-data/interfaith_DDL.sql (hunk 2 above)
Comment (Tiernebre) [reply docker/postgres/docker-entrypoint-initdb.d/mock-data/interfaith_DDL.sql:61]: Yep. My assumption was that, let's say someone creates an alert, but doesn't want to send it out immediately. They can send it out at a later time by sending it. We can clarify this at the next meeting if this is what an alert should do.

Comment (TylerNicholls) [reply docker/postgres/docker-entrypoint-initdb.d/mock-data/interfaith_DDL.sql:61]: I agree, though it seems like a question of if Jody wants to have that functionality or if she just wants to push an alert when she presses a button and doesn't care about scheduling future ones. I could really see her going either way on it

**Pull request merged**